modules/nf-core/sourmash/taxannotate/main.nf

```diff
@@ -0,0 +1,43 @@
+process SOURMASH_TAXANNOTATE {
+    tag "$meta.id"
+    label 'process_single'
+
+    conda (params.enable_conda ? "bioconda::sourmash=4.6.1" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/sourmash:4.6.1--hdfd78af_0':
+        'quay.io/biocontainers/sourmash:4.6.1--hdfd78af_0' }"
+
+    input:
+    tuple val(meta), path(gather_results)
+    path(taxonomy)
+
+    output:
+    tuple val(meta), path("*.with-lineages.csv.gz"), emit: result
+    path "versions.yml"                            , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args   = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+
+    sourmash \\
+        tax annotate \
+        $args \\
+        --gather-csv ${gather_results} \\
+        --taxonomy ${taxonomy} \\
+        --output-dir "."
+
+    ## Output file name = ${prefix}.with-lineages.csv
+
+    ## Compress output
+    gzip "${prefix}.with-lineages.csv"
```

Comment:
I don't fully follow this - how does `sourmash` know to use `prefix` when it is not specifyied in the command?Is there some magic here? (otherwise I don't think the `## output file` name comment is necessary, to me (at least) it's self explanatory :)

Reply:
ya there is magic here -- `sourmash tax annotate` chops `.csv` off of the gather file name and adds `.with-lineages.csv` to the end of it. (i don't have an opinion about the `## Output file name` comment

Reply:
That's true, it looks cryptic.
Previously, as sourmash was not able to read gzipped-input, we unpacked input from `gather` and named it with `meta.id`, so it was clear. So now the process expects that `meta.id` is the same as `gather_results` (as inherited from `SOURMASH_GATHER`).

To fix it, maybe we can explicitly get the input file name and put it into a variable `prefix` inside the script itself? 
Please see the latest commit: https://github.com/nf-core/modules/pull/2422/commits/f85f107b9434279496247f4f66edf07fd2fcc0d7

Reply:
Hm, Ok. It still seems a bit over-complicated for me.

What about:

`gzip -cvf ${gather_results.getBaseName}.with-lineages.csv > ${prefix}.with-lineages.csv` [^1]

That way you can use exactly the output file name in the same way as sourmash (just using the native nextflow method), and that way you don't need any bash variables etc.

[^1]: getBaseName from Nextflow file attributes: https://www.nextflow.io/docs/latest/script.html#check-file-attributes

Reply:
Sure, we can move `prefix` extraction to the Nextflow side.
However, I think we can not use just `getBaseName` as we need to remove both extensions (csv.gz), and it's not safe to use `getSimpleName` as it can remove part of the sample name if it contains period.
Also I tried to avoid unpacking the date into temporary file, as sourmash can read gz-comressed data.
So I've changed the code and used `prefix="${gather_results.getBaseName().replaceAll(/.csv/, '')}"`

Reply:
Ah fair point!